chia-bls/src/signature.rs

```diff
@@ -283,12 +403,27 @@ pub fn sign<Msg: AsRef<[u8]>>(sk: &SecretKey, msg: Msg) -> Signature {
     Signature(p2)
 }
 
+pub fn sign<Msg: AsRef<[u8]>>(sk: &SecretKey, msg: Msg) -> Signature {
```

Comment:
Wouldn't it make sense for this to be a method of `SecretKey` since you always need one to sign a message?

Reply:
perhaps it would. this is modelled after blspy where there are 3 different ways of signing a message (aug, pop and basic). Chia only uses Aug, so that's the only supported mechanism in `chia-bls`.

In general I think there's a tendency in the world to make free functions members, because that's what Java does, or maybe that's what their IDE suggests. You also always need a message to sign with a key, so maybe it should be a member of the message? I'm not seriously suggesting that, but I bet some would.